fenix/app/src/main/java/org/mozilla/fenix/HomeActivityNavDirections.kt

```diff
@@ -0,0 +1,163 @@
+import org.mozilla.fenix.library.recentlyclosed.RecentlyClosedFragmentDirections
+import org.mozilla.fenix.search.SearchDialogFragmentDirections
+import org.mozilla.fenix.settings.HttpsOnlyFragmentDirections
+import org.mozilla.fenix.settings.SettingsFragmentDirections
+import org.mozilla.fenix.settings.TrackingProtectionFragmentDirections
+import org.mozilla.fenix.settings.about.AboutFragmentDirections
+import org.mozilla.fenix.settings.logins.fragment.LoginDetailFragmentDirections
+import org.mozilla.fenix.settings.logins.fragment.SavedLoginsAuthFragmentDirections
+import org.mozilla.fenix.settings.search.SaveSearchEngineFragmentDirections
+import org.mozilla.fenix.settings.search.SearchEngineFragmentDirections
+import org.mozilla.fenix.settings.studies.StudiesFragmentDirections
+import org.mozilla.fenix.settings.wallpaper.WallpaperSettingsFragmentDirections
+import org.mozilla.fenix.share.AddNewDeviceFragmentDirections
+import org.mozilla.fenix.shopping.ReviewQualityCheckFragmentDirections
+import org.mozilla.fenix.tabstray.TabsTrayFragmentDirections
+import org.mozilla.fenix.trackingprotection.TrackingProtectionPanelDialogFragmentDirections
+import java.security.InvalidParameterException
+
+/**
+ * Delegate to the relevant get nav directions function based on the given [Activity].
+ *
+ * @param from The [BrowserDirection] to indicate which fragment the browser is being opened from.
+ * @param customTabSessionId Optional custom tab session ID if navigating from a custom tab.
+ *
+ * @return the [NavDirections] for the given [Activity].
+ */
+fun Activity.getNavDirections(
+    from: BrowserDirection,
+    customTabSessionId: String? = null,
+): NavDirections? = when (this) {
+    is ExternalAppBrowserActivity -> {
+        getExternalAppBrowserNavDirections(from, customTabSessionId)
+    }
+
+    else -> {
+        getHomeNavDirections(from)
+    }
+}
+
+private fun Activity.getExternalAppBrowserNavDirections(
+    from: BrowserDirection,
+    customTabSessionId: String?,
+): NavDirections? {
+    if (customTabSessionId == null) {
+        finishAndRemoveTask()
+        return null
+    }
+
+    val manifest = intent
+        .getWebAppManifest()
+        ?.let { WebAppManifestParser().serialize(it).toString() }
+    return when (from) {
+        BrowserDirection.FromGlobal ->
+            NavGraphDirections.actionGlobalExternalAppBrowser(
+                activeSessionId = customTabSessionId,
+                webAppManifest = manifest,
+                isSandboxCustomTab = intent.getBooleanExtra(EXTRA_IS_SANDBOX_CUSTOM_TAB, false),
+            )
+
+        else -> throw InvalidParameterException(
+            "Tried to navigate to ExternalAppBrowserFragment from $from",
+        )
+    }
+}
+
+private fun getHomeNavDirections(
+    from: BrowserDirection,
+): NavDirections = when (from) {
+    BrowserDirection.FromGlobal ->
+        NavGraphDirections.actionGlobalBrowser(null)
```

Comment:
how come these directions no longer need `customTabSessionId`?

Reply:
> the home directions never needed it in the first place?
That's exactly it 👍 

> Is it because of the when inside of Activity.getNavDirections
Also yes, so the idea being that the `Activity`s don't have responsibility in determining which `NavDirections` are required

Reply:
^ can we add this to a comment above this method. It looks suspicious and would save a future reader some time if the 'why' was described.

Reply:
Nice suggestion @jonalmeida, instead I've updated the parameter default to null so we don't have to do this anymore 🙂